README.md
# AIW-Azure-Network-Solutions


LabFiles/Release-notes.md
# Release Notes

### 12 July 2022

### Azure-Network-Solutions

In this lab, you will learn how to setup and configure a virtual network with subnets in Azure. You will deploy a pre-configured IaaS environment then redesign and update it to account for high availability with the help of Azure Load balancer. You will provision Virtual Network in a secondary region and deploy a virtual machine within the Vnet. You'll also configure Peering connections between the Virtual Networks and test inter-site Virtual Machine connectivity on your private network.

You will learn how to enhance the security of your application with the help of Azure application gateway (WAF) and Azure Firewall. You will configure User-defined route (UDR) and outbound traffic to go via Azure Firewall. You will enable Network Watcher and utilize tools to monitor, troubleshoot, view metrics, and enable logs for resources in your Azure virtual network.

This lab includes the following exercises:

### Exercise 1: Getting Started with Azure 

In this exercise, you will log in to the Azure Portal and review the pre-deployed resource groups that are deployed as part of the lab environment.

### Exercise 2: Provision Network Landing Zone

In this exercise, you'll create a Virtual Network and subnets for different scenarios like AVD, DMZ, Internal, and AzureFirewallSubnet.

### Exercise 3: Provision Contoso Insurance Application in Existing Network

In this exercise, you will deploy a Virtual Machine using a pre-built ARM template, create a Public IP address and associate it to the Virtual Machine. Then you'll create a network security group, add rules to allow outbound traffic, and associate it with the virtual machine's network interface. You will then test the Contoso Insurance application.

### Exercise 4:  Setup Load Balancing 

In this exercise, you will provision a secondary Virtual Machine with the Contoso Insurance application hosted on it. You will deploy an external Load Balancer, place Virtual Machines in the backend pool and check the load balancing between virtual machines. You will also test the behavior of the load balancer in case of VM failover.

### Exercise 5: Provision & Configure Azure Firewall & Application Gateway

In this exercise, you will deploy a Firewall and Application Gateway with WAF then you will publish an application through them. You'll also test the application for security and perform a sample attack, and also you'll configure UDR and outbound traffic to go via Azure Firewall.

### Exercise 6: Provision Secondary Azure Region Infrastructure

In this exercise, you'll create a Virtual Network in a secondary region and deploy a VM within the Vnet. You'll also configure Peering connections between the Vnets and test inter-VM connectivity on your private network.

### Exercise 7: Monitoring Azure Network

In this exercise, you will enable a Network Watcher and create Flow Logs to run Sample traffic and perform Sample network troubleshooting. You'll also deploy Log Analytics Workspace and Network Monitoring Solution to set up Azure Monitor Alerts and test notifications.


template/psscript.ps1
Param (
    [Parameter(Mandatory = $true)]
    [string]
    $location
)

Start-Transcript -Path C:\WindowsAzure\Logs\CloudLabsCustomScriptExtension.txt -Append
# To resolve the error of https://github.com/microsoft/MCW-App-modernization/issues/68. The cause of the error is Powershell by default uses TLS 1.0 to connect to website, but website security requires TLS 1.2. You can change this behavior with running any of the below command to use all protocols. You can also specify single protocol.
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls, [Net.SecurityProtocolType]::Tls11, [Net.SecurityProtocolType]::Tls12, [Net.SecurityProtocolType]::Ssl3
[Net.ServicePointManager]::SecurityProtocol = "Tls, Tls11, Tls12, Ssl3"

Install-WindowsFeature -Name "web-server" `
                       -IncludeManagementTools `
                       -IncludeAllSubFeature

#Download and Unpack the Website
$zipDownload = "https://github.com/CloudLabsAI-Azure/AIW-Azure-Network-Solutions/blob/main/template/WebApp.zip?raw=true"
#$zipDownload = "https://www.dropbox.com/s/wcfnuf76h3tn3ws/ContosoInsuranceIIS.zip?dl=1"
$downloadedFile = "C:\ContosoInsuranceIIS.zip"
$inetpubFolder = "C:\inetpub\wwwroot"
Invoke-WebRequest -Uri  $zipDownload -OutFile $downloadedFile
Add-Type -assembly "system.io.compression.filesystem"
[io.compression.zipfile]::ExtractToDirectory($downloadedFile, $inetpubFolder)

$srvname = "vm-contoso-frontend-prod-eastus-001"

(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "computernamevalue", "$srvname"} | Set-Content -Path "C:\inetpub\wwwroot\Home.html"
(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "azureregionvalue", "$location"} | Set-Content -Path "C:\inetpub\wwwroot\Home.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "azureregionvalue", "$location"} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "computernamevalue", "$srvname"} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "Sample text. replace the text with some story line", "Contoso Insurance (CI), headquartered in Miami, provides insurance solutions across North America. Its products include accident and health insurance, life insurance, travel, home, and auto coverage. <br> <br> Contoso insures that they have the right breadth and depth of continuity protection.  Contoso will evenly distribute the load across the backend servers to serve the request in time. Based on the demand, contoso will scale the application and create highly available services to handle the requests efficiently. <br> <br> In case of a lights-out event, they will be able to control the systems failover with zero downtime."} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "Sample text. replace the text with some story line", "Contoso Insurance (CI), headquartered in Miami, provides insurance solutions across North America. Its products include accident and health insurance, life insurance, travel, home, and auto coverage. <br> <br> Contoso insures that they have the right breadth and depth of continuity protection.  Contoso will evenly distribute the load across the backend servers to serve the request in time. Based on the demand, contoso will scale the application and create highly available services to handle the requests efficiently. <br> <br> In case of a lights-out event, they will be able to control the systems failover with zero downtime."} | Set-Content -Path "C:\inetpub\wwwroot\index.html"


Function InstallEdgeChromium
{
    #Download and Install edge
    $WebClient = New-Object System.Net.WebClient
    $WebClient.DownloadFile("https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/59c478d3-513a-4060-837b-01ad385d6aaa/MicrosoftEdgeEnterpriseX86.msi","C:\Packages\MicrosoftEdgeBetaEnterpriseX64.msi")
    sleep 5
    
    Start-Process msiexec.exe -Wait '/I C:\Packages\MicrosoftEdgeBetaEnterpriseX64.msi /qn' -Verbose 
    sleep 5
    $WshShell = New-Object -comObject WScript.Shell
    $Shortcut = $WshShell.CreateShortcut("C:\Users\Public\Desktop\Azure Portal.lnk")
    $Shortcut.TargetPath = """C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe"""
    $argA = """https://portal.azure.com"""
    $Shortcut.Arguments = $argA 
    $Shortcut.Save()

}
InstallEdgeChromium


template/psscript3.ps1
Param (
    [Parameter(Mandatory = $true)]
    [string]
    $location
)

Start-Transcript -Path C:\WindowsAzure\Logs\CloudLabsCustomScriptExtension.txt -Append
# To resolve the error of https://github.com/microsoft/MCW-App-modernization/issues/68. The cause of the error is Powershell by default uses TLS 1.0 to connect to website, but website security requires TLS 1.2. You can change this behavior with running any of the below command to use all protocols. You can also specify single protocol.
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls, [Net.SecurityProtocolType]::Tls11, [Net.SecurityProtocolType]::Tls12, [Net.SecurityProtocolType]::Ssl3
[Net.ServicePointManager]::SecurityProtocol = "Tls, Tls11, Tls12, Ssl3"

Install-WindowsFeature -Name "web-server" `
                       -IncludeManagementTools `
                       -IncludeAllSubFeature

#Download and Unpack the Website
$zipDownload = "https://github.com/CloudLabsAI-Azure/AIW-Azure-Network-Solutions/blob/main/template/WebApp.zip?raw=true"
#$zipDownload = "https://www.dropbox.com/s/wcfnuf76h3tn3ws/ContosoInsuranceIIS.zip?dl=1"
$downloadedFile = "C:\ContosoInsuranceIIS.zip"
$inetpubFolder = "C:\inetpub\wwwroot"
Invoke-WebRequest -Uri  $zipDownload -OutFile $downloadedFile
Add-Type -assembly "system.io.compression.filesystem"
[io.compression.zipfile]::ExtractToDirectory($downloadedFile, $inetpubFolder)

$srvname = "vm-contoso-app-prod-westus-001"

(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "computernamevalue", "$srvname"} | Set-Content -Path "C:\inetpub\wwwroot\Home.html"
(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "azureregionvalue", "$location"} | Set-Content -Path "C:\inetpub\wwwroot\Home.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "azureregionvalue", "$location"} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "computernamevalue", "$srvname"} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "Sample text. replace the text with some story line", "Contoso Insurance (CI), headquartered in Miami, provides insurance solutions across North America. Its products include accident and health insurance, life insurance, travel, home, and auto coverage. <br> <br> Contoso insures that they have the right breadth and depth of continuity protection.  Contoso will evenly distribute the load across the backend servers to serve the request in time. Based on the demand, contoso will scale the application and create highly available services to handle the requests efficiently. <br> <br> In case of a lights-out event, they will be able to control the systems failover with zero downtime."} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "Sample text. replace the text with some story line", "Contoso Insurance (CI), headquartered in Miami, provides insurance solutions across North America. Its products include accident and health insurance, life insurance, travel, home, and auto coverage. <br> <br> Contoso insures that they have the right breadth and depth of continuity protection.  Contoso will evenly distribute the load across the backend servers to serve the request in time. Based on the demand, contoso will scale the application and create highly available services to handle the requests efficiently. <br> <br> In case of a lights-out event, they will be able to control the systems failover with zero downtime."} | Set-Content -Path "C:\inetpub\wwwroot\index.html"


Function InstallEdgeChromium
{
    #Download and Install edge
    $WebClient = New-Object System.Net.WebClient
    $WebClient.DownloadFile("https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/59c478d3-513a-4060-837b-01ad385d6aaa/MicrosoftEdgeEnterpriseX86.msi","C:\Packages\MicrosoftEdgeBetaEnterpriseX64.msi")
    sleep 5
    
    Start-Process msiexec.exe -Wait '/I C:\Packages\MicrosoftEdgeBetaEnterpriseX64.msi /qn' -Verbose 
    sleep 5
    $WshShell = New-Object -comObject WScript.Shell
    $Shortcut = $WshShell.CreateShortcut("C:\Users\Public\Desktop\Azure Portal.lnk")
    $Shortcut.TargetPath = """C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe"""
    $argA = """https://portal.azure.com"""
    $Shortcut.Arguments = $argA 
    $Shortcut.Save()

}
InstallEdgeChromium


template/psscript2.ps1
Param (
    [Parameter(Mandatory = $true)]
    [string]
    $location
)

Start-Transcript -Path C:\WindowsAzure\Logs\CloudLabsCustomScriptExtension.txt -Append
# To resolve the error of https://github.com/microsoft/MCW-App-modernization/issues/68. The cause of the error is Powershell by default uses TLS 1.0 to connect to website, but website security requires TLS 1.2. You can change this behavior with running any of the below command to use all protocols. You can also specify single protocol.
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls, [Net.SecurityProtocolType]::Tls11, [Net.SecurityProtocolType]::Tls12, [Net.SecurityProtocolType]::Ssl3
[Net.ServicePointManager]::SecurityProtocol = "Tls, Tls11, Tls12, Ssl3"

Install-WindowsFeature -Name "web-server" `
                       -IncludeManagementTools `
                       -IncludeAllSubFeature

#Download and Unpack the Website
$zipDownload = "https://github.com/CloudLabsAI-Azure/AIW-Azure-Network-Solutions/blob/main/template/WebApp.zip?raw=true"
#$zipDownload = "https://www.dropbox.com/s/wcfnuf76h3tn3ws/ContosoInsuranceIIS.zip?dl=1"
$downloadedFile = "C:\ContosoInsuranceIIS.zip"
$inetpubFolder = "C:\inetpub\wwwroot"
Invoke-WebRequest -Uri  $zipDownload -OutFile $downloadedFile
Add-Type -assembly "system.io.compression.filesystem"
[io.compression.zipfile]::ExtractToDirectory($downloadedFile, $inetpubFolder)

$srvname = "vm-contoso-frontend-prod-eastus-002"

(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "computernamevalue", "$srvname"} | Set-Content -Path "C:\inetpub\wwwroot\Home.html"
(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "azureregionvalue", "$location"} | Set-Content -Path "C:\inetpub\wwwroot\Home.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "azureregionvalue", "$location"} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "computernamevalue", "$srvname"} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\index.html") | ForEach-Object {$_ -Replace "Sample text. replace the text with some story line", "Contoso Insurance (CI), headquartered in Miami, provides insurance solutions across North America. Its products include accident and health insurance, life insurance, travel, home, and auto coverage. <br> <br> Contoso insures that they have the right breadth and depth of continuity protection.  Contoso will evenly distribute the load across the backend servers to serve the request in time. Based on the demand, contoso will scale the application and create highly available services to handle the requests efficiently. <br> <br> In case of a lights-out event, they will be able to control the systems failover with zero downtime."} | Set-Content -Path "C:\inetpub\wwwroot\index.html"
(Get-Content -Path "C:\inetpub\wwwroot\Home.html") | ForEach-Object {$_ -Replace "Sample text. replace the text with some story line", "Contoso Insurance (CI), headquartered in Miami, provides insurance solutions across North America. Its products include accident and health insurance, life insurance, travel, home, and auto coverage. <br> <br> Contoso insures that they have the right breadth and depth of continuity protection.  Contoso will evenly distribute the load across the backend servers to serve the request in time. Based on the demand, contoso will scale the application and create highly available services to handle the requests efficiently. <br> <br> In case of a lights-out event, they will be able to control the systems failover with zero downtime."} | Set-Content -Path "C:\inetpub\wwwroot\index.html"


Function InstallEdgeChromium
{
    #Download and Install edge
    $WebClient = New-Object System.Net.WebClient
    $WebClient.DownloadFile("https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/59c478d3-513a-4060-837b-01ad385d6aaa/MicrosoftEdgeEnterpriseX86.msi","C:\Packages\MicrosoftEdgeBetaEnterpriseX64.msi")
    sleep 5
    
    Start-Process msiexec.exe -Wait '/I C:\Packages\MicrosoftEdgeBetaEnterpriseX64.msi /qn' -Verbose 
    sleep 5
    $WshShell = New-Object -comObject WScript.Shell
    $Shortcut = $WshShell.CreateShortcut("C:\Users\Public\Desktop\Azure Portal.lnk")
    $Shortcut.TargetPath = """C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe"""
    $argA = """https://portal.azure.com"""
    $Shortcut.Arguments = $argA 
    $Shortcut.Save()

}
InstallEdgeChromium
